src/sidebar/components/datalist.js

```diff
@@ -0,0 +1,100 @@
+const { createElement } = require('preact');
+const classnames = require('classnames');
+const propTypes = require('prop-types');
+const { useMemo } = require('preact/hooks');
+
+/**
+ * Custom datalist component. Use this in conjunction with an input field.
+ */
+
+function Datalist({
+  activeItem = -1,
+  id,
+  itemPrefixId,
+  list,
+  listFormatter = item => item,
```

Comment:
Note that since default arguments are evaluated on every function call in JavaScript, `listFormatter` will be different every time if not passed as a prop, this will invalidate the `useMemo` dependencies below. A way to avoid that is to hoist the value to a top-level constant:

```js
const defaultListFormatter = item => item;

function Datalist({
...,
listFormatter = defaultListFormatter,
...
})
```